### ISSUE TITLE
compilation error

### ISSUE THREAD

**Issue by gamask** [serge-sans-paille/pythran#1561]
```python
#pythran export get_taxable (
#   float64 [],
#   int, int,
#   int, int, int,
#   float64 list, float64 list,
#   float64 [], float64 [],
#   float64, float64,
#   int list, int list, int list, int list,
#   float64 [][] order(C), bool [], float64 [],
#   float64 [], float64 [][] order(C), float64 [][] order(C), float64 [][] order(C), float64 [][] order(C), float64 [][] order(C)
#)
def get_taxable (
    x,
    b, c,
    n_free_vars, y, cnum,
    g, h,
    balance, interest_rates,
    cpp, oas,
    nonreg_indices, rif_indices, corp_indices, lif_indices,
    other_income, split_pension, corp_cda_init,
    nonreg_acb_init, nonreg_bond_percent, nonreg_interest, nonreg_eligible_dividends, nonreg_distributed_gains, nonreg_advisory_fee
):
    nonreg  = x[n_free_vars * y + nonreg_indices[cnum]]
    reg     = x[n_free_vars * y + rif_indices[cnum]]
    corp    = x[n_free_vars * y + corp_indices[cnum]]
    lif     = x[n_free_vars * y + lif_indices[cnum]]
    
    pension_db = g[y] if cnum == 0 else h[y]
    
    ret = 0
    ret += cpp
    ret += oas
    
    if split_pension[y]:
        ret += 123.
    else:
        ret += pension_db + lif + reg
    
    ret += other_income[y][cnum]
    
    nonreg_balance = 123.
    
    
    if nonreg_balance > 0.:
        nonreg_acb = 123.
        ret += nonreg * (1 - nonreg_acb / nonreg_balance) / 2.
    else:
        ret += nonreg * (1                              ) / 2.
    
    ret += corp * 1.15      # CCC
    
    ret += nonreg_balance * nonreg_bond_percent[y][cnum] * nonreg_interest[y][cnum]
    
    ret += nonreg_balance * 123. * nonreg_eligible_dividends[y][cnum] * 1.38
    
    ret += nonreg_balance * 123. * nonreg_distributed_gains[y][cnum] / 2.
    
    ret -= nonreg_balance * nonreg_advisory_fee[y][cnum]
    
    corp_cda = 123.
    
    # AAA
    if corp_cda > 0.:
        if corp < corp_cda:
            ret -= corp
        else:
            ret -= corp_cda
    else:
        pass
    
    # BBB
    #ret -= corp_cda
    
    return max (ret, 0)

```

Code as is fails to compile and fails to give a reason (gives generic error message involving Jaaaack).
Comment out line marked CCC and error goes away.
Alternatively uncomment line BBB and comment out block AAA and error also goes away.

**Comment by gamask** [comment]
Works fine on windows though.

**Comment by serge-sans-paille** [comment]
And works fine on Linux on with master branch for me...

**Comment by gamask** [comment]
I'm looking into it. Confirmed that I'm indeed on master, and reinstalled pythran, and it's still happening.
If running with -E the generated cpp code is identical between linux and windows, with the exception of some include statements that are in a different order.

**Comment by gamask** [comment]
Fails with g++ 5.5.0-12ubuntu1~16.04
Fails with g++ 6.5.0-2ubuntu1~16.04
Succeeds with g++ 7.5.0-3ubuntu1~16.04
Succeeds with g++ 9.3.0-10ubuntu2~16.04

With g++ 5 and 6 I was seeing enormous RAM consumption. "top" was reporting over 90% use, at which point it, and the rest of the system, became unresponsive for a few seconds. Then the compilation would error out, usually. Sometimes it would freeze for at least a minute, at which point I restarted my computer. I have 16 gigabytes of RAM.

With g++ 7 and 9 RAM consumption (of cc1plus) hovers around 20%.
Compilation speed is comparable between these two versions.

Code as is takes 80 seconds to compile.
With line CCC commented out it takes 38 seconds.
With block AAA commented and line BBB uncommented it takes 29 seconds.
On my windows machine, which is slower, the unmodified code takes under 5 seconds to compile. It's using MSVC 14.23.28105.

My actual code (as opposed to this stripped-down example) takes a few minutes to compile, and probably under 20 seconds on Windows. Before I added this get_taxable function the entire code was compiling just as quickly on linux.
